docs/howto/script_testing.md

```diff
@@ -0,0 +1,209 @@
+
+# Start the service.
+docker_signal test-hits SIGHUP
+
+# Wait for the service to exit.
+docker_wait_exit -timeout 5m test-hits
+
+# Check that we can see our policy.
+get_policy -profile ${CONFIG_PROFILES}/${PROFILE} -timeout 1m ${DATA_STREAM_NAME}
+cp stdout got_policy.json
+exec jq '.name=="'${DATA_STREAM_NAME}'"' got_policy.json
+stdout true
+
+# Take down the service and check logs for our message.
+docker_down test-hits
+! stderr .
+stdout '"total_lines":10'
+
+# Get documents from the data stream.
+get_docs -profile ${CONFIG_PROFILES}/${PROFILE} -want 10 -timeout 5m ${DATA_STREAM_NAME}
+cp stdout got_docs.json
+
+# Remove the data stream.
+remove_package_policy -profile ${CONFIG_PROFILES}/${PROFILE} ${DATA_STREAM_NAME}
+
+# Uninstall the agent.
+uninstall_agent -profile ${CONFIG_PROFILES}/${PROFILE} -timeout 1m 
+
+# Remove the package resources.
+remove_package -profile ${CONFIG_PROFILES}/${PROFILE}
+
+-- test-hits/docker-compose.yml --
+version: '2.3'
+services:
+  test-hits:
+    image: docker.elastic.co/observability/stream:v0.20.0
+    volumes:
+      - ./logs:/logs:ro
+    command: log --start-signal=SIGHUP --delay=5s --addr elastic-agent:9999 -p=tcp /logs/generated.log
+-- test-hits/logs/generated.log --
+ntpd[1001]: kernel time sync enabled utl
+restorecond: : Reset file context quasiarc: liqua
+auditd[5699]: Audit daemon rotating log files
+anacron[5066]: Normal exit ehend
+restorecond: : Reset file context vol: luptat
+heartbeat: : <<eumiu.medium> Processing command: accept
+restorecond: : Reset file context nci: ofdeFin
+auditd[6668]: Audit daemon rotating log files
+anacron[1613]: Normal exit mvolu
+ntpd[2959]: ntpd gelit-r tatno
+-- test_config.yaml --
+input: tcp
+vars: ~
+data_stream:
+  vars:
+    tcp_host: 0.0.0.0
+    tcp_port: 9999
+```
+
+Other complete examples can be found in the [with_script test package](https://github.com/elastic/elastic-package/blob/main/test/packages/other/with_script/data_stream/first/_dev/test/scripts).
+
+
+## Running script tests
+
+The `elastic-package test script` command has the following sub-command-specific
+flags:
+
+- `--continue`: continue running the script if an error occurs
+- `--data-streams`: comma-separated data streams to test
+- `--external-stack`: use external stack for script tests (default true)
+- `--run`: run only tests matching the regular expression
+- `--scripts`: path to directory containing test scripts (advanced use only)
+- `--update`: update archive file if a cmp fails
+- `--verbose-scripts`: verbose script test output (show all script logging)
+- `--work`: print temporary work directory and do not remove when done
+
+
+## Limitations
+
+While the testscript package allows reference to paths outside the configuration
+root and the package's root, the backing `elastic-package` infrastructure does
+not, so it is advised that tests only refer to paths within the `$WORK` and
+`$PKG_ROOT` directories.
```

Comment:
Looking into the environment variables listed above, should this be `PACKAGE_ROOT`?
```suggestion
`$PACKAGE_ROOT` directories.
```